src/ui-fragment/src/types.ts

```diff
@@ -3,3 +3,11 @@ export interface CssClasses {
 	name: string;
 	content: string;
 }
+
+export interface Action {
+	source: number;
+	signal: string;
+	destination: number;
+	slot: string;
+	slotParam: string | number | boolean;
```

Comment:
Shouldn't this be an array of `any`? With the idea being that `slotParams` is the list of parameters passed to the slot function.